From 47c420297ddefdc482507eadd49090a4ff49e6d9 Mon Sep 17 00:00:00 2001
From: Danesh M <daneshm90@gmail.com>
Date: Fri, 14 Aug 2015 11:20:31 -0700
Subject: [PATCH 36/74] FMRadio : Ensure scan state is kept in sync

In the middle of a scan, the user could navigate away from the activity,
during which the scan could've finished. The activity is no longer synchronized
with the service and causes issues where the dialog for scan keeps showing...etc

Fix this by syncing state in onResume

issue-id: CRACKLING-529
Change-Id: Ief3138e4d6e157ad0b060dc3aced18668959f4e1
---
 fmapp2/src/com/caf/fmradio/FMRadio.java         | 14 ++++++++++++++
 fmapp2/src/com/caf/fmradio/FMRadioService.java  | 10 ++++++++++
 fmapp2/src/com/caf/fmradio/IFMRadioService.aidl |  1 +
 3 files changed, 25 insertions(+)

diff --git a/fmapp2/src/com/caf/fmradio/FMRadio.java b/fmapp2/src/com/caf/fmradio/FMRadio.java
index 0cc8e71..18242ce 100644
--- a/fmapp2/src/com/caf/fmradio/FMRadio.java
+++ b/fmapp2/src/com/caf/fmradio/FMRadio.java
@@ -524,11 +524,25 @@ public class FMRadio extends Activity
       }
    }
 
+    private void syncScanState() {
+        if (!mIsScaning || mService == null) {
+            return;
+        }
+        try {
+            if (!mService.isSearchInProgress()) {
+                resetSearch();
+            }
+        }catch (RemoteException e) {
+            e.printStackTrace();
+        }
+    }
+
    @Override
    public void onResume() {
       Log.d(LOGTAG, "FMRadio: onResume");
 
       super.onResume();
+      syncScanState();
 
       if (mService == null) {
           Log.d(LOGTAG,"bind callback has not received yet - wait for 100ms");
diff --git a/fmapp2/src/com/caf/fmradio/FMRadioService.java b/fmapp2/src/com/caf/fmradio/FMRadioService.java
index 5ce91ae..0169f3d 100644
--- a/fmapp2/src/com/caf/fmradio/FMRadioService.java
+++ b/fmapp2/src/com/caf/fmradio/FMRadioService.java
@@ -2267,6 +2267,11 @@ public class FMRadioService extends Service
          return(mService.get().isA2DPConnected());
       }
 
+      public boolean isSearchInProgress()
+      {
+         return(mService.get().isSearchInProgress());
+      }
+
       public int getExtenCountryCode()
       {
          return(mService.get().getExtenCountryCode());
@@ -2731,6 +2736,11 @@ public class FMRadioService extends Service
       return(bStatus);
    }
 
+   public boolean isSearchInProgress() {
+      int state = mReceiver.getFMState();
+      return state == qcom.fmradio.FmTransceiver.FMState_Srch_InProg;
+   }
+
    public boolean isSSRInProgress() {
       return mIsSSRInProgress;
    }
diff --git a/fmapp2/src/com/caf/fmradio/IFMRadioService.aidl b/fmapp2/src/com/caf/fmradio/IFMRadioService.aidl
index 07d210e..3359542 100644
--- a/fmapp2/src/com/caf/fmradio/IFMRadioService.aidl
+++ b/fmapp2/src/com/caf/fmradio/IFMRadioService.aidl
@@ -76,6 +76,7 @@ interface IFMRadioService
     boolean isSSRInProgress();
     boolean isRtPlusSupported();
     boolean isA2DPConnected();
+    boolean isSearchInProgress();
     boolean getIntfDetLowTh();
     boolean getIntfDetHighTh();
     boolean getRxRepeatCount();
-- 
2.17.1

